### PR TITLE
feat: [CDS-51737]: Added the trigger support for bamboo

### PR DESCRIPTION
**Pull request by shivamnegi94**
## Describe your changes
Added the trigger support for bamboo Artifact source.

Test Cases:

- Verified the Trigger creation is successful.
- Verified the Polling for the Bamboo Artifact is successful
Screenshot 2023-03-23 at 9 12 17 AM
Screenshot 2023-03-23 at 9 12 49 AM
Screenshot 2023-03-23 at 9 13 40 AM

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/45027)

### CODE REVIEW

**Comment by shivamnegi94** [comment]
trigger ut8

**Review by vtxorxwitty — DISMISSED** [review]
> Reviewed 25 of 30 files at r1, 4 of 4 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45027)* status: 30 file(s) to review. 1 of 2 approvals obtained. Waiting on team(s): del-team.

**Review by vtxorxwitty — DISMISSED** [review]
> Reviewed 6 of 6 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45027)* status: 30 file(s) to review. 1 of 2 approvals obtained. Waiting on team(s): del-team.

File: 260-delegate/config-delegate.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ grpcServiceConnectorPort: 8080
 logStreamingServiceBaseUrl: http://localhost:8079
 clientToolsDownloadDisabled: false
 installClientToolsInBackground: true
-isLocalNgDelegate: false
+isLocalNgDelegate: true
```
Comment (sribalij) [inline 260-delegate/config-delegate.yml:19]: Can we remove this?

File: 260-delegate/src/test/resources/kryo-registrations.txt [hunk 2]
```diff
@@ -2744,4 +2743,4 @@
 10000518:io.harness.delegate.task.aws.lambda.AwsLambdaArtifactoryArtifactConfig
 10000519:io.harness.delegate.task.aws.lambda.AwsLambdaCustomArtifactConfig
 10000520:io.harness.delegate.task.pcf.request.CfRouteMappingRequestNG
-10000521:io.harness.delegate.task.pcf.response.CfRouteMappingResponseNG
\ No newline at end of file
+10000521:io.harness.delegate.task.pcf.response.CfRouteMappingResponseNG
```
Comment (sribalij) [inline 260-delegate/src/test/resources/kryo-registrations.txt:2746]: Do you still need this?

**Review by sribalij — COMMENTED** [review]
Can you also inlcude unit tests?

File: 260-delegate/config-delegate.yml (hunk 1 above)
Comment (shivamnegi94) [reply 260-delegate/config-delegate.yml:19]: I will revert

File: 260-delegate/src/test/resources/kryo-registrations.txt (hunk 2 above)
Comment (shivamnegi94) [reply 260-delegate/src/test/resources/kryo-registrations.txt:2746]: Rebase added the changes

**Comment by shivamnegi94** [comment]
trigger ut2

**Review by sribalij — APPROVED** [review]
> Reviewed 2 of 4 files at r2, 1 of 1 files at r3, 2 of 6 files at r4, 1 of 25 files at r5, 17 of 24 files at r6, 6 of 6 files at r7, 5 of 5 files at r8, 2 of 2 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45027)* status: 31 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained. Waiting on team(s): del-team.

**Review by vtxorxwitty — APPROVED** [review]
> Reviewed 1 of 25 files at r5, 17 of 24 files at r6, 6 of 6 files at r7, 5 of 5 files at r8, 2 of 2 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45027)* status: 2. unresolved discussions. Waiting on team(s): del-team.

**Comment by shivamnegi94** [comment]
trigger messagecheck

**Review by sribalij — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45027)* status: 2. unresolved discussions. Waiting on team(s): del-team.

**Review by sribalij — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45027)* status: Waiting on team(s): del-team.

**Review by raghuAtWings — APPROVED** [review]
> Reviewed 2 of 4 files at r2, 1 of 1 files at r3, 2 of 6 files at r4, 1 of 25 files at r5, 17 of 24 files at r6, 6 of 6 files at r7, 5 of 5 files at r8, 2 of 2 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45027)* status: Waiting on team(s): del-team.

**Pull request merged**